library/modules/bigip_gtm_dns_listener.py

```diff
@@ -0,0 +1,836 @@
+        'address',
+        'port',
+        'advertise',
+        'auto_lasthop',
+        'state',
+        'ip_protocol',
+        'fallback_persistence',
+        'translate_address',
+        'translate_port',
+        'source_port',
+        'vlans',
+        'last_hop_pool',
+        'rules',
+        'vlans_enabled',
+        'vlans_disabled'
+    ]
+
+    @property
+    def state(self):
+        if self._values['state'] == 'enabled':
+            return 'present'
+        return self._values['state']
+
+    @property
+    def enabled(self):
+        if self._values['enabled'] is None:
+            return None
+        return True
+
+    @property
+    def disabled(self):
+        if self._values['disabled'] is None:
+            return None
+        return True
+
+
+class ApiParameters(Parameters):
+    pass
+
+
+class ModuleParameters(Parameters):
+    @property
+    def description(self):
+        if self._values['description'] is None:
+            return None
+        elif self._values['description'] in ['none', '']:
+            return ''
+        return self._values['description']
+
+    @property
+    def address(self):
+        if self._values['address'] is None:
+            return None
+        return self._values['address']
+
+    @property
+    def port(self):
+        if self._values['port'] is None:
+            return None
+        return self._values['port']
+
+    @property
+    def ip_protocol(self):
+        if self._values['ip_protocol'] is None:
+            return None
+        return self._values['ip_protocol']
+
+    @property
+    def mask(self):
+        if self._values['mask'] is None:
+            return None
+        return self._values['mask']
+
+    @property
+    def translate_address(self):
+        if self._values['translate_address'] is None:
+            return None
+        return self._values['translate_address']
+
+    @property
+    def translate_port(self):
+        if self._values['translate_port'] is None:
+            return None
+        return self._values['translate_port']
+
+    @property
+    def advertise(self):
+        if self._values['advertise'] is None:
+            return None
+        return flatten_boolean(self._values['advertise'])
+
+    @property
+    def vlans_enabled(self):
+        if self._values['vlans_enabled'] is None:
+            return None
+        elif self._values['vlans_enabled'] is True:
+            return True
+        return False
+
+    @property
+    def vlans_disabled(self):
+        if self._values['vlans_disabled'] is None:
+            return None
+        elif self._values['vlans_disabled'] is True:
+            return True
+        return False
+
+    @property
+    def vlans(self):
+        if self._values['vlans'] is None:
+            return None
+        if is_empty_list(self._values['vlans']):
+            return []
+        result = list()
+        for item in self._values['vlans']:
+            result.append(item)
+        return result
```

Comment:
why do you append list to a list? what is the purpose

Reply:
I agree - this is unnecessary for loop and it was removed to return just vlans value. 

Thanks for pointing out to the existent logic in virtual server module! 

I have implemented the same way as it is done for virtual server. 

This comment is addressed now.